### ISSUE TITLE
Issue when building with intel compilers icc (ICC) 2021.6.0 20220226

### ISSUE THREAD

**Issue by janjust** [openpmix/openpmix#2795]
Tried building OpenMPI main branch with interal pmix (pmix master).

gcc seems to work fine, but when building with intel compiler I get the following error:

```
make[4]: Entering directory '/global/scratch/users/tomislavj/ompi-dev/ompi-build-main/3rd-party/openpmix/src/util'
  CC       pmix_alfg.lo
  CC       pmix_argv.lo
  CC       pmix_cmd_line.lo
  CC       pmix_error.lo
  CC       pmix_printf.lo
  CC       pmix_output.lo
  CC       pmix_environ.lo
  CC       pmix_fd.lo
  CC       pmix_timings.lo
  CC       pmix_os_dirpath.lo
  CC       pmix_os_path.lo
  CC       pmix_basename.lo
  CC       pmix_keyval_parse.lo
  CC       pmix_show_help.lo
  CC       pmix_path.lo
  CC       pmix_getid.lo
  CC       pmix_shmem.lo
  CC       pmix_vmem.lo
  CC       pmix_hash.lo
  CC       pmix_name_fns.lo
  CC       pmix_net.lo
  CC       pmix_if.lo
  CC       pmix_parse_options.lo
  CC       pmix_context_fns.lo
  CC       pmix_pty.lo
  CC       pmix_few.lo
  CC       pmix_string_copy.lo
  CC       pmix_getcwd.lo
/global/scratch/users/tomislavj/ompi-dev/ompi/3rd-party/openpmix/src/util/pmix_keyval_parse.c(103): error #188: enumerated type mixed with another type
                  parse_line_new(filename, val, callback);
                                           ^

/global/scratch/users/tomislavj/ompi-dev/ompi/3rd-party/openpmix/src/util/pmix_keyval_parse.c(328): error #188: enumerated type mixed with another type
              val = pmix_util_keyval_yylex();
                  ^

/global/scratch/users/tomislavj/ompi-dev/ompi/3rd-party/openpmix/src/util/pmix_keyval_parse.c(347): error #188: enumerated type mixed with another type
              val = pmix_util_keyval_yylex();
                  ^

/global/scratch/users/tomislavj/ompi-dev/ompi/3rd-party/openpmix/src/util/pmix_keyval_parse.c(364): error #188: enumerated type mixed with another type
          val = pmix_util_keyval_yylex();
              ^

compilation aborted for /global/scratch/users/tomislavj/ompi-dev/ompi/3rd-party/openpmix/src/util/pmix_keyval_parse.c (code 2)
make[4]: *** [Makefile:887: pmix_keyval_parse.lo] Error 1
make[4]: *** Waiting for unfinished jobs....
```

configure option was: 
` --prefix=<path> --with-pmix=internal --with-slurm --with-libevent=internal --enable-mca-no-build=coll-ml,btl-uct,btl-openib CC=icc CXX=icpc F77=ifort FC=ifort`

**Comment by rhc54** [comment]
I vaguely recall fixing something like that a while back - I have no idea where OMPI main is with respect to PMIx master.

**Comment by awlauria** [comment]
@rhc54 ompi `main` is pointing to the head of PMIx main at this time.

**Comment by rhc54** [comment]
Okay - I can provide a first-cut at it, but perhaps someone can then tidy it up if there are lingering problems? I have no access to that compiler and really don't want to install it on my limited system.

**Comment by rhc54** [comment]
Fixed by #2796 . @awlauria Can you please update OMPI main and then have @janjust try again?
